packages/shared/friends/sagas.ts

```diff
@@ -75,13 +78,25 @@ function* initializeFriendsSaga() {
   if (identity.hasConnectedWeb3) {
     yield call(waitForRealmInitialized)
 
-    try {
-      const synapseUrl: string = yield select(getSynapseUrl)
-      yield call(initializePrivateMessaging, synapseUrl, identity)
-    } catch (e) {
-      logger.error(`error initializing private messaging`, e)
+    let secondsToRetry = MIN_TIME_BETWEEN_FRIENDS_INITIALIZATION_RETRIES_MILLIS
+
+    while(true) {
+      try {
+        const synapseUrl: string = yield select(getSynapseUrl)
+        yield call(initializePrivateMessaging, synapseUrl, identity)
```

Comment:
when does this loop ends?

Reply:
but it doesn't stop if it worked, or I am missing it

Reply:
Here are two examples of controlled infinite loops https://github.com/decentraland/kernel/blob/chore/new-comms-sagas/packages/shared/comms/sagas.ts#L114-L154

Reply:
I'd add something like 
```ts
const client: SocialAPI | null = yield select(getClient)
const isLoggedIn: boolean = client && (yield apply(client, client.isLoggedIn) || false
if (isLoggedIn) {
  // reset timers
  // yield delay(initialDelay)
} else {
  yield call(initializePrivateMessaging, synapseUrl, identity)
}
```